Remove unused files & add MacOS instructions

cpeel opened the pull request:
This MR does two notable things:
1. Removes unused perl modules
2. Adds instructions for how to make Guiguts work on MacOS

cpeel commented:
@srjfoo - I would appreciate validation of the MacOS instructions. These instructions seemed to work for me.

srjfoo commented:
I'll be happy to!

cpeel commented:
Unless you also installed Tk::Toolbar from CPAN into your perl installation, this is expected. Guiguts still needs that and Tk::LineNumberText -- we just want it to get and use the CPAN version instead of bundling it in.

windymilla commented:
Yep. That makes sense

On Sun, 7 Jun 2020, 16:25 Casey Peel, <notifications@github.com> wrote:

> Unless you also installed Tk::Toolbar from CPAN into your perl
> installation, this is expected. Guiguts still needs that and
> Tk::LineNumberText -- we just want it to get and use the CPAN version
> instead of bundling it in.
>
> —
> You are receiving this because your review was requested.
> Reply to this email directly, view it on GitHub
> <https://github.com/DistributedProofreaders/guiguts/pull/42#issuecomment-640235466>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ACEGIVPP74KZ5M2L6VT25ITRVOWQLANCNFSM4NWYT6OA>
> .
>

cpeel commented:
I removed the commit that removed the CPAN modules. I want to push that in with another commit that address some of the larger Windows/perl question.

windymilla commented:
Not important, but maybe of interest: my GG is running happily without lib/Tk/LineNumberText, though it does have lib/Guiguts/LineNumberText (looks like an edited/stripped down version)

cpeel commented:
> Not important, but maybe of interest: my GG is running happily without lib/Tk/LineNumberText, though it does have lib/Guiguts/LineNumberText (looks like an edited/stripped down version)

Indeed - it works for me without that too. I've removed it from the list of needed modules.

This branch is now rebased off of HEAD.

cpeel commented:
After more testing (and with help from @srjfoo) I've flushed out the [MacOS instructions](https://github.com/cpeel/guiguts/blob/remove-unused-files/INSTALL.md#macos) more. I've also added a helper script to install the perl modules which we should be able to reuse on Windows as well.

srjfoo commented on INSTALL.md:
```diff
@@ -0,0 +1,140 @@
+
+* LWP::UserAgent
+* Tk
+* Tk::ToolBar
```
`HTML::TokeParser` isn't actually needed, after all? (That seems to have disappeared form the instructions.)

cpeel replied:
It is needed but as several people have pointed out, it is included with the base perl install for both Windows and Mac so I opted not to call it out (like we don't call out all the other modules needed but included with perl).

srjfoo replied:
Makes sense -- I hadn't realized it was part of the base perl install, just noticed the difference between these instructions and the earlier ones.

cpeel commented on INSTALL.md:
```diff
@@ -0,0 +1,86 @@
+theory these should not be overwritten).
+
+Please direct any help requests to the
+[DP forum Guiguts thread](http://www.pgdp.net/phpBB2/viewtopic.php?t=3075).
```
Yes, I'll get that updated.

cpeel commented on INSTALL.md:
````diff
@@ -0,0 +1,86 @@
+
+Now unzip `guiguts-n.n.n.zip` and start Guiguts with:
+```
+perl guiguts.pl
````
Also a good idea!

cpeel commented:
We've had some good reports on the Mac instructions, so I think this is good for merging pending OK by @windymilla.

windymilla commented:
Yes, happy with this
